src/renderer/image_renderer.rs

```diff
@@ -103,8 +121,21 @@ impl ImageRenderer<Static> {
         }
 
         ffi::MapRenderer_setCamera(self.instance.pin_mut(), lat, lon, zoom, bearing, pitch);
-        let img = ffi::MapRenderer_render(self.instance.pin_mut());
-        Ok(Image(img))
+        let data = ffi::MapRenderer_render(self.instance.pin_mut());
+        let bytes = data.as_bytes();
+
+        // Parse dimensions from first 8 bytes
+        if bytes.len() < 8 {
+            return Err(RenderingError::InvalidImageData);
+        }
+
+        let width = u32::from_le_bytes([bytes[0], bytes[1], bytes[2], bytes[3]]);
```

Comment:
here, you assume `le` ordering, whereas in C++ you just push it with `data.append(reinterpret_cast<const char*>(&width), sizeof(uint32_t));` -- this might be the source of problems in the future. Can you return a Rust struct instead?  Note that the c++ can use rust structs directly because of cxx